### PR TITLE
Jeload.cpp refactoring

### PR DESCRIPTION
**Pull request by ahmadbeirkdar**
Refactored jeload.cpp.
What's left is void* pointers which need to be done when other files are being refactored.

Added helper functions for strings, in util/

### CODE REVIEW

**Comment by codereport** [comment]
Are you able to try and clean up the history (maybe `git rebase`)?

**Comment by ahmadbeirkdar** [comment]
> Are you able to try and clean up the history (maybe `git rebase`)?

All done, my bad

File: util/stringx.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+#ifndef J_STRINGX_H
+#define J_STRINGX_H
```
[the remark this answers is not in the document]
Comment (ahmadbeirkdar) [reply util/stringx.h:3]: For sure, it’s not a library, I wrote it as a “utility” function which came in handy and should be reusable around the code base

[the remark this answers is not in the document]
Comment (ahmadbeirkdar) [reply util/stringx.h]: Named it stringx to not conflict with std::string, and to store all helper string functions

File: jsrc/util/stringx.h [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+#pragma once
+
+#include <string>
+
+namespace util{
+    auto remove_all_occurrences(std::string &str, std::string_view const &sub_str) -> void;
```
Comment (codereport) [inline jsrc/util/stringx.h:7]: `std::string_view` doesn't need to be `const&` because it is non-owning :slightly_smiling_face:

**Review by codereport — CHANGES_REQUESTED** [review]

File: jsrc/CMakeLists.txt [hunk 3]
```diff
@@ -224,6 +224,8 @@ target_sources(jconsole PRIVATE
     jeload.h
     jeload.cpp
     jlib.h
+        util/stringx.cpp
+        util/stringx.h
```
Comment (codereport) [inline jsrc/CMakeLists.txt:228]: indentation

File: jsrc/jeload.cpp [hunk 4]
```diff
@@ -2,233 +2,214 @@
 /* Licensed use only. Any other use is in violation of copyright.          */
 // utilities for JFE to load JE, initiallize, and run profile sentence
 // JFEs are jconsole, jwdw, and jwdp
+// TODO: Remove all void* where applicable when other parts of the code start being refactored
 #include <string>
 #include <stdexcept>
+#include "util/stringx.h"
```
Comment (codereport) [inline jsrc/jeload.cpp:8]: split standard and user headers

File: jsrc/jeload.h [hunk 5]
```diff
@@ -3,15 +3,16 @@
...
 
+
 void jepath(char*,char *);
 void jesetpath(char*);
 void* jeload(void* callbacks); // returns J
-int jefirst(int,char*);
 A jegeta(int,char*);
 I jeseta(I n,char* name,I x,char* d);
-int jedo(char*); // could be jedi
+int jedo(const char*); // could be jedi
```
Comment (codereport) [inline jsrc/jeload.h:12]: east const

File: jsrc/util/stringx.cpp [hunk 6]
```diff
@@ -0,0 +1,11 @@
+
```
Comment (codereport) [inline jsrc/util/stringx.cpp:1]: EOF newline

File: jsrc/jeload.cpp (hunk 4 above)
Comment (codereport) [inline jsrc/jeload.cpp:8]: Also, i think it is fine to rename to `string.hpp/cpp`

**Review by codereport — CHANGES_REQUESTED** [review]
We can merge after this.

**Review by codereport — APPROVED** [review]
lgtm :+1:

**Pull request merged**